src/NLog/LogEventInfo.cs

```diff
@@ -281,20 +280,75 @@ public string FormattedMessage
         /// <summary>
         /// Gets the dictionary of per-event context properties.
         /// </summary>
-        public IDictionary<object, object> Properties
+        public IDictionary<object,object> Properties
         {
             get
             {
                 if (this.properties == null)
                 {
-                    this.properties = new Dictionary<object, object>();
+                    this.properties = new PropertiesDictionary();
                 }
 
                 return this.properties;
             }
         }
 
         /// <summary>
+        /// 
+        /// </summary>
+        public IMessageTemplate MessageTemplate
+        {
+            get
+            {
+                bool positional = true;
+                IList<MessageTemplateParameter> messageTemplateParameters = ArrayHelper.Empty<MessageTemplateParameter>();
+                if (this.properties != null && this.properties.MessageProperties != null && this.properties.MessageProperties.Count > 0)
+                {
+                    messageTemplateParameters = this.properties.MessageProperties;
+                    if (this.properties.MessageProperties[0].Name != null)
+                    {
+                        if (this.properties.MessageProperties[0].Name.Length != 1 || this.properties.MessageProperties[0].Name[0] < '0' || this.properties.MessageProperties[0].Name[0] > '9')
```

Comment:
why is this needed in the getter? For performance reasons?

Reply:
Will remove it, since it will never happen

Reply:
Fixed